File: src/src_core/fmg/core/mosaic/MosaicImageController.cs
using System;
using fmg.common;
using fmg.core.img;

namespace fmg.core.mosaic {

   /// <summary> MVC: mosaic image controller. Base implementation </summary>
   public abstract class MosaicImageController<TImage,          TMosaicView>
                            : MosaicController<TImage, Nothing, TMosaicView, MosaicAnimatedModel<Nothing>>,
                            IAnimatedController<TImage,          TMosaicView, MosaicAnimatedModel<Nothing>>
      where TImage : class
      where TMosaicView : MosaicView<TImage, Nothing, MosaicAnimatedModel<Nothing>>
   {

      private readonly AnimatedInnerController<TImage, TMosaicView, MosaicAnimatedModel<Nothing>> _innerController;

      public MosaicImageController(TMosaicView view)
         : base(view)
      {
         var model = Model;
         _innerController = new AnimatedInnerController<TImage, TMosaicView, MosaicAnimatedModel<Nothing>>(model);
         AddModelTransformer(new MosaicRotateTransformer<Nothing>());

         var pen = model.PenBorder;
         pen.ColorLight = pen.ColorShadow;
         model.BkFill.Mode = 1 + ThreadLocalRandom.Current.Next(model.CellAttr.GetMaxBackgroundFillModeValue());
      }

      public void AddModelTransformer(IModelTransformer transformer) {
         _innerController.AddModelTransformer(transformer);
      }
      public void RemoveModelTransformer(Type /* extends IModelTransformer */ transformerClass) {
         _innerController.RemoveModelTransformer(transformerClass);
      }

   }

}

File: src/src_core/fmg/core/mosaic/MosaicImageController.java
package fmg.core.mosaic;

import java.util.concurrent.ThreadLocalRandom;

import fmg.core.img.*;
import fmg.core.types.draw.PenBorder;

/** MVC: mosaic image controller. Base implementation */
public abstract class MosaicImageController<TImage,        TMosaicView extends MosaicView<TImage, Void, MosaicAnimatedModel<Void>>>
                   extends MosaicController<TImage, Void,  TMosaicView,                                 MosaicAnimatedModel<Void>>
             implements IAnimatedController<TImage,        TMosaicView,                                 MosaicAnimatedModel<Void>>
{
    private final AnimatedInnerController<TImage, TMosaicView, MosaicAnimatedModel<Void>> _innerController;

    public MosaicImageController(TMosaicView view) {
        super(view);
        MosaicAnimatedModel<Void> model = getModel();
        _innerController = new AnimatedInnerController<>(model);
        addModelTransformer(new MosaicRotateTransformer());

        PenBorder pen = model.getPenBorder();
        pen.setColorLight(pen.getColorShadow());
        model.getBackgroundFill().setMode(1 + ThreadLocalRandom.current().nextInt(model.getCellAttr().getMaxBackgroundFillModeValue()));
    }

    @Override
    public void addModelTransformer(IModelTransformer transformer) {
        _innerController.addModelTransformer(transformer);
    }
    @Override
    public void removeModelTransformer(Class<? extends IModelTransformer> transformerClass) {
        _innerController.removeModelTransformer(transformerClass);
    }

}
